pkg/minishift/openshift/version/openshift_versions.go

```diff
@@ -88,36 +73,43 @@ func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
 	for _, tag := range tagsList {
 		fmt.Fprintf(output, "\t- %s\n", tag)
 	}
+	return nil
 }
 
-func PrintUpStreamVersions(output io.Writer, minSupportedVersion string) {
-	resp, err := getResponseBody("https://registry.hub.docker.com/v2/repositories/openshift/origin/tags/")
+func PrintUpStreamVersions(output io.Writer, minSupportedVersion string, defaultVersion string) error {
+	dockerRegistryUrl := "https://registry.hub.docker.com/v1/repositories/openshift/origin/tags"
+	resp, err := getResponseBody(dockerRegistryUrl)
 	if err != nil {
-		fmt.Fprintf(output, "Error Occured %s", err)
-		return
+		return err
 	}
 	defer resp.Body.Close()
 	decoder := json.NewDecoder(resp.Body)
-	var data ImageTags
+	var data []ImageTags
 	err = decoder.Decode(&data)
 	if err != nil {
-		fmt.Printf("%T\n%s\n%#v\n", err, err, err)
-		return
+		return errors.New(fmt.Sprintf("%T\n%s\n%#v\n", err, err, err))
 	}
 	fmt.Fprint(output, "The following OpenShift versions are available: \n")
 	var tagsList []string
-	for _, imageinfo := range data.Results {
+	for _, imageinfo := range data {
 		if strings.Contains(imageinfo.Name, "latest") {
 			continue
 		}
-		if util.VersionOrdinal(imageinfo.Name) >= util.VersionOrdinal(minSupportedVersion) {
-			tagsList = append(tagsList, imageinfo.Name)
+		if valid, _ := clusterup.ValidateOpenshiftMinVersion(imageinfo.Name, minSupportedVersion); valid {
```

Comment:
imo, ValidateOpenshiftMinVersion should have been moved into another package. It feels wrong to have a dependency to the clusterup package here. Also, ValidateOpenshiftMinVersion does not really tell what this method is doing. But anyways, ...